Realm/Realm/DatabaseTypes/RealmValue.cs

```diff
@@ -1533,8 +1533,6 @@ public bool Equals(RealmValue other)
                 RealmValueType.ObjectId => AsObjectId() == other.AsObjectId(),
                 RealmValueType.Guid => AsGuid() == other.AsGuid(),
                 RealmValueType.Object => AsIRealmObject().Equals(other.AsIRealmObject()),
-                RealmValueType.List => AsList().SequenceEqual(other.AsList()),
-                RealmValueType.Dictionary => AsDictionary().DictionaryEquals(other.AsDictionary()),
                 RealmValueType.Null => true,
```

Comment:
This looks a bit suspicious 🤨

Reply:
The `Null` part? It got me for a second too 😁 
Before the switch we check that the type of the two RealmValues is the same. So if they're both null they're automatically equal.

Reply:
Ah, makes sense